syntax="proto3";

package cog;

message MarketData {
  optional string f_5051 = 1;
  optional double f_2 = 2;
  optional int32 f_3 = 3;
  optional int64 f_100 = 4;
  optional double f_302 = 5;
  optional double f_304 = 6;
  optional double f_301 = 7;
  optional double f_300 = 8;
  optional double f_5 = 9;
  optional int32 f_7 = 10;
  optional double f_4 = 11;
  optional int32 f_6 = 12;
  optional double f_18 = 13;
  optional double f_118 = 14;
  optional double f_106 = 15;
  optional int32 f_308 = 16;
  optional int32 f_5610 = 17;
  optional int32 f_5611 = 18;
  optional int32 f_5612 = 19;
  optional int32 f_5613 = 20;
  optional int32 f_5614 = 21;
  optional int32 f_5595 = 22;
  optional int32 f_5596 = 23;
  optional int32 f_5597 = 24;
  optional int32 f_5598 = 25;
  optional int32 f_5599 = 26;
  optional double f_221 = 27;
  optional double f_222 = 28;
  optional double f_223 = 29;
  optional double f_224 = 30;
  optional double f_225 = 31;
  optional double f_226 = 32;
  optional double f_227 = 33;
  optional double f_228 = 34;
  optional double f_229 = 35;
  optional double f_230 = 36;
  optional int32 f_231 = 37;
  optional int32 f_232 = 38;
  optional int32 f_233 = 39;
  optional int32 f_234 = 40;
  optional int32 f_235 = 41;
  optional int32 f_236 = 42;
  optional int32 f_237 = 43;
  optional int32 f_238 = 44;
  optional int32 f_239 = 45;
  optional int32 f_240 = 46;
  optional double f_201 = 47;
  optional double f_202 = 48;
  optional double f_203 = 49;
  optional double f_204 = 50;
  optional double f_205 = 51;
  optional double f_206 = 52;
  optional double f_207 = 53;
  optional double f_208 = 54;
  optional double f_209 = 55;
  optional double f_210 = 56;
  optional int32 f_211 = 57;
  optional int32 f_212 = 58;
  optional int32 f_213 = 59;
  optional int32 f_214 = 60;
  optional int32 f_215 = 61;
  optional int32 f_216 = 62;
  optional int32 f_217 = 63;
  optional int32 f_218 = 64;
  optional int32 f_219 = 65;
  optional int32 f_220 = 66;
  optional int32 f_5602 = 67;
  optional int32 f_5603 = 68;
  optional int32 f_5604 = 69;
  optional int32 f_5605 = 70;
  optional int32 f_5606 = 71;
  optional int32 f_5590 = 72;
  optional int32 f_5591 = 73;
  optional int32 f_5592 = 74;
  optional int32 f_5593 = 75;
  optional int32 f_5594 = 76;
  optional double f_5864 = 77;
  optional string f_5620 = 78;
  optional string f_5939 = 79;
  optional double f_10 = 80;
  optional double f_11 = 81;
  optional double f_12 = 82;
  optional string ts = 83;
  optional int64 msg_offset = 84;
  optional double f_112 = 85;
  optional double f_23 = 86;
  optional double f_497 = 87;
  optional double f_8019 = 88;
  optional double f_111 = 89;
  optional double f_24 = 90;
  optional double f_7219 = 91;
  optional double f_8673 = 92;
  optional string f_481 = 93;
  optional string f_1084 = 94;
  optional string f_762 = 95;
  optional double f_129 = 96;
  optional double f_757 = 97;
  optional string f_438 = 98;
  optional double f_5894 = 99;
  optional double f_8675 = 100;
  optional double f_8667 = 101;
  optional double f_5833 = 102;
  optional double f_21 = 103;
  optional double f_8679 = 104;
  optional double f_8665 = 105;
  optional string f_403 = 106;
  optional string f_5112 = 107;
  optional string f_5350 = 108;
  optional double f_5355 = 109;
  optional double f_5893 = 110;
  optional double f_5930 = 111;
  optional double f_5933 = 112;
  optional int32 f_10012 = 113;
  optional int32 message_latency = 114;
  optional string f_400 = 115;
  optional string f_5105 = 116;
  optional string f_5104 = 117;
  optional double f_316 = 118;
  optional string f_5121 = 119;
  optional string f_8103 = 120;
  optional double f_418 = 121;
  optional double f_415 = 122;
  optional int32 f_411 = 123;
  optional double f_416 = 124;
  optional double f_417 = 125;
  optional string f_5066 = 126;
  optional int32 f_25010 = 127;
  optional double f_34 = 128;
  optional double f_2160 = 129;
  optional int32 f_467 = 130;
  optional double f_5632 = 131;
  optional double f_5631 = 132;
  optional double f_5144 = 133;
  optional double f_5146 = 134;
  optional double f_5148 = 135;
  optional double f_5425 = 136;
  optional double f_5173 = 137;
  optional double f_5152 = 138;
  optional double f_5427 = 139;
  optional double f_5154 = 140;
  optional double f_9548 = 141;
  optional double f_6589 = 142;
  optional double f_5170 = 143;
  optional double f_5128 = 144;
  optional double f_5130 = 145;
  optional double f_5132 = 146;
  optional double f_5002 = 147;
  optional double f_5052 = 148;
  optional double f_5054 = 149;
  optional double f_8760 = 150;
  optional double f_5133 = 151;
  optional double f_5003 = 152;
  optional double f_5053 = 153;
  optional double f_8761 = 154;
  optional double f_25100 = 155;
  optional string topic = 156;
  optional bool recap = 157;
  optional int32 f_7010 = 158;
  optional double f_6675 = 159;
  optional string f_6678 = 160;
  optional double f_5301 = 161;
  optional double f_414 = 162;
  optional string f_5093 = 163;
  optional double f_5964 = 164;
  optional string f_6840 = 165;
  optional string f_9911 = 166;
  optional string f_5502 = 167;
  optional double f_412 = 168;
  optional double f_5414 = 169;
  optional int32 f_20077 = 170;
  optional double f_13148 = 171;
  optional double f_5129 = 172;
  optional double f_5415 = 173;
  optional double f_5416 = 174;
  optional double f_5417 = 175;
  optional double f_5418 = 176;
  optional double f_5419 = 177;
  optional double f_5420 = 178;
  optional double f_5421 = 179;
  optional int64 cum_vol = 180;
  optional double cum_pv = 181;
  optional double vwap = 182;
  optional string f_9 = 183;
  optional double f_116 = 184;
  optional string f_200 = 185;
  optional int32 f_309 = 186;
  optional double f_322 = 187;
  optional double f_323 = 188;
  optional double f_334 = 189;
  optional int64 f_336 = 190;
  optional double f_338 = 191;
  optional double f_432 = 192;
  optional double f_474 = 193;
  optional double f_496 = 194;
  optional string f_5056 = 195;
  optional string f_5113 = 196;
  optional double f_5124 = 197;
  optional double f_5150 = 198;
  optional double f_5201 = 199;
  optional double f_5202 = 200;
  optional double f_5204 = 201;
  optional double f_5209 = 202;
  optional double f_5210 = 203;
  optional double f_5215 = 204;
  optional double f_5218 = 205;
  optional double f_5219 = 206;
  optional double f_5226 = 207;
  optional double f_5237 = 208;
  optional double f_5240 = 209;
  optional double f_5241 = 210;
  optional double f_5244 = 211;
  optional double f_5245 = 212;
  optional double f_5246 = 213;
  optional double f_5248 = 214;
  optional double f_5263 = 215;
  optional double f_5264 = 216;
  optional double f_5273 = 217;
  optional double f_5274 = 218;
  optional string f_5297 = 219;
  optional double f_5401 = 220;
  optional double f_5402 = 221;
  optional double f_5403 = 222;
  optional double f_5404 = 223;
  optional double f_5405 = 224;
  optional double f_5406 = 225;
  optional double f_5407 = 226;
  optional double f_5408 = 227;
  optional double f_5409 = 228;
  optional double f_5410 = 229;
  optional double f_5411 = 230;
  optional double f_5412 = 231;
  optional double f_5413 = 232;
  optional double f_5422 = 233;
  optional double f_5423 = 234;
  optional double f_5424 = 235;
  optional double f_5426 = 236;
  optional string f_5691 = 237;
  optional int64 f_5986 = 238;
  optional double f_5989 = 239;
  optional double f_5993 = 240;
  optional int64 f_5998 = 241;
  optional double f_5999 = 242;
  optional string f_6057 = 243;
  optional string f_6058 = 244;
  optional double f_6166 = 245;
  optional double f_6167 = 246;
  optional double f_6168 = 247;
  optional double f_6169 = 248;
  optional double f_6193 = 249;
  optional double f_6194 = 250;
  optional double f_6195 = 251;
  optional double f_6196 = 252;
  optional double f_6197 = 253;
  optional double f_6198 = 254;
  optional double f_6199 = 255;
  optional double f_6236 = 256;
  optional double f_6237 = 257;
  optional double f_6238 = 258;
  optional double f_6239 = 259;
  optional double f_6240 = 260;
  optional double f_6241 = 261;
  optional double f_6242 = 262;
  optional double f_6243 = 263;
  optional double f_6244 = 264;
  optional double f_6245 = 265;
  optional double f_6246 = 266;
  optional double f_6247 = 267;
  optional double f_6248 = 268;
  optional string f_6252 = 269;
  optional double f_6345 = 270;
  optional double f_6346 = 271;
  optional double f_6347 = 272;
  optional double f_6348 = 273;
  optional double f_6351 = 274;
  optional double f_6352 = 275;
  optional double f_6353 = 276;
  optional double f_6354 = 277;
  optional double f_6355 = 278;
  optional double f_6356 = 279;
  optional double f_6357 = 280;
  optional double f_6358 = 281;
  optional string f_6362 = 282;
  optional string f_6363 = 283;
  optional string f_6364 = 284;
  optional string f_6365 = 285;
  optional string f_6366 = 286;
  optional string f_6367 = 287;
  optional double f_6368 = 288;
  optional double f_6369 = 289;
  optional double f_6372 = 290;
  optional double f_6373 = 291;
  optional double f_6374 = 292;
  optional double f_6375 = 293;
  optional double f_6376 = 294;
  optional double f_6377 = 295;
  optional double f_6454 = 296;
  optional double f_6603 = 297;
  optional double f_6604 = 298;
  optional double f_6605 = 299;
  optional double f_6606 = 300;
  optional double f_6607 = 301;
  optional double f_6700 = 302;
  optional string f_6783 = 303;
  optional string f_6784 = 304;
  optional string f_6802 = 305;
  optional string f_6830 = 306;
  optional string f_7001 = 307;
  optional double f_7014 = 308;
  optional string f_7024 = 309;
  optional string f_7025 = 310;
  optional string f_7026 = 311;
  optional string f_7027 = 312;
  optional string f_7028 = 313;
  optional double f_7037 = 314;
  optional double f_7038 = 315;
  optional double f_7039 = 316;
  optional double f_7074 = 317;
  optional string f_7141 = 318;
  optional string f_7142 = 319;
  optional double f_7147 = 320;
  optional double f_7148 = 321;
  optional double f_7149 = 322;
  optional double f_7150 = 323;
  optional double f_7151 = 324;
  optional double f_7152 = 325;
  optional double f_7153 = 326;
  optional double f_7154 = 327;
  optional double f_7155 = 328;
  optional double f_7156 = 329;
  optional double f_7158 = 330;
  optional double f_7159 = 331;
  optional double f_7160 = 332;
  optional double f_7162 = 333;
  optional double f_7163 = 334;
  optional double f_7164 = 335;
  optional double f_7165 = 336;
  optional double f_7166 = 337;
  optional double f_7167 = 338;
  optional double f_7168 = 339;
  optional double f_7169 = 340;
  optional double f_7170 = 341;
  optional double f_7171 = 342;
  optional double f_7172 = 343;
  optional double f_7173 = 344;
  optional double f_7174 = 345;
  optional double f_7175 = 346;
  optional double f_7176 = 347;
  optional double f_7177 = 348;
  optional double f_7178 = 349;
  optional double f_7179 = 350;
  optional double f_7180 = 351;
  optional double f_7181 = 352;
  optional double f_7183 = 353;
  optional double f_7184 = 354;
  optional double f_7185 = 355;
  optional double f_7186 = 356;
  optional double f_7187 = 357;
  optional double f_7188 = 358;
  optional double f_7189 = 359;
  optional double f_7190 = 360;
  optional double f_7191 = 361;
  optional double f_7192 = 362;
  optional double f_7193 = 363;
  optional double f_7194 = 364;
  optional double f_7195 = 365;
  optional int32 f_7196 = 366;
  optional double f_7197 = 367;
  optional double f_7199 = 368;
  optional string f_8101 = 369;
  optional string f_8115 = 370;
  optional string f_8150 = 371;
  optional double f_8750 = 372;
  optional double f_8752 = 373;
  optional double f_8753 = 374;
  optional double f_8755 = 375;
  optional double f_8756 = 376;
  optional double f_8757 = 377;
  optional double f_8758 = 378;
  optional string f_8766 = 379;
  optional double f_8767 = 380;
  optional double f_9079 = 381;
  optional double f_9080 = 382;
  optional double f_9081 = 383;
  optional double f_9082 = 384;
  optional double f_9083 = 385;
  optional double f_9084 = 386;
  optional double f_9085 = 387;
  optional double f_9086 = 388;
  optional double f_9087 = 389;
  optional double f_9088 = 390;
  optional string f_9101 = 391;
  optional string f_9102 = 392;
  optional int64 f_9145 = 393;
  optional int32 f_9155 = 394;
  optional double f_9161 = 395;
  optional int32 f_9265 = 396;
  optional double f_9306 = 397;
  optional string f_9718 = 398;
  optional double f_10004 = 399;
  optional int64 f_13114 = 400;
  optional int64 f_13115 = 401;
  optional string f_13118 = 402;
  optional string f_26000 = 403;
  optional string f_26001 = 404;
  optional string f_26002 = 405;
  optional string f_26003 = 406;
  optional string f_26005 = 407;
  optional string f_26006 = 408;
  optional string f_26007 = 409;
  optional int32 f_26008 = 410;
  optional string f_26009 = 411;
  optional string f_26010 = 412;
  optional string f_26011 = 413;
  optional string f_26012 = 414;
  optional double f_26013 = 415;
  optional double f_26014 = 416;
  optional double f_26015 = 417;
  optional double f_26016 = 418;
  optional double f_26017 = 419;
  optional double f_26018 = 420;
  optional double f_26019 = 421;
  optional double f_26020 = 422;
  optional double f_26021 = 423;
  optional double f_26022 = 424;
  optional double f_26023 = 425;
  optional double f_26024 = 426;
  optional double f_26025 = 427;
  optional double f_26026 = 428;
  optional double f_26027 = 429;
  optional double f_26028 = 430;
  optional double f_26029 = 431;
  optional double f_26030 = 432;
  optional double f_26031 = 433;
  optional double f_26032 = 434;
  optional double f_26033 = 435;
  optional double f_26034 = 436;
  optional double f_26035 = 437;
  optional double f_26036 = 438;
  optional double f_26037 = 439;
  optional double f_26038 = 440;
  optional double f_26039 = 441;
  optional double f_26040 = 442;
  optional double f_26041 = 443;
  optional double f_26042 = 444;
  optional double f_26043 = 445;
  optional double f_26044 = 446;
  optional double f_26045 = 447;
  optional double f_26046 = 448;
  optional double f_26047 = 449;
  optional double f_26048 = 450;
  optional double f_26049 = 451;
  optional double f_26050 = 452;
  optional double f_26051 = 453;
  optional double f_26052 = 454;
  optional double f_26053 = 455;
  optional double f_26054 = 456;
  optional double f_26055 = 457;
  optional double f_26056 = 458;
  optional double f_26057 = 459;
  optional double f_26058 = 460;
  optional double f_26059 = 461;
  optional double f_26060 = 462;
  optional double f_26061 = 463;
  optional double f_26062 = 464;
  optional double f_26063 = 465;
  optional double f_26064 = 466;
  optional double f_26065 = 467;
  optional double f_26066 = 468;
  optional double f_26067 = 469;
  optional double f_26068 = 470;
  optional double f_26069 = 471;
  optional double f_26070 = 472;
  optional double f_26071 = 473;
  optional double f_26072 = 474;
  optional double f_26073 = 475;
  optional double f_26074 = 476;
  optional double f_26075 = 477;
  optional double f_26076 = 478;
  optional double f_26077 = 479;
  optional double f_26078 = 480;
  optional double f_26079 = 481;
  optional double f_26080 = 482;
  optional double f_26081 = 483;
  optional double f_26082 = 484;
  optional double f_26083 = 485;
  optional double f_26084 = 486;
  optional double f_26085 = 487;
  optional double f_26086 = 488;
  optional double f_26087 = 489;
  optional double f_26088 = 490;
  optional double f_26089 = 491;
}

message NewsItem {
  string story_id =1;
  string headline =2;
  string story =3;
  string symbols=4;
  string isins=5;
  string categories=6;
  string sources = 7;
  double epoch_millis = 8;
}

enum MarketDataUpdateType {
    UPDATE = 0;
    SNAPSHOT = 1;
}

message MarketDataBatch{
  repeated MarketData items = 1;
}

message MarketDataUpdate {
  MarketDataUpdateType type =1;
    oneof payload {
    MarketData market_data = 2;
    MarketDataBatch market_data_batch = 3;
    NewsItem news_item =4;
  }
}

